var/spack/repos/builtin/packages/h5hut/package.py

```diff
@@ -47,7 +48,11 @@ def validate(self):
 
     def flag_handler(self, name, flags):
         build_system_flags = []
-        if name == "cflags" and self.spec["hdf5"].satisfies("@1.12:"):
+        if (
+            name == "cflags"
+            and self.spec.satisfies("@:2")
```

Comment:
With [h5hut@2.0.0rc7](https://gitlab.psi.ch/H5hut/src/-/tags/H5hut-2.0.0rc7) the release note says

> fixes to support hdf5 >= 1.12 with new API calls

And actually in the [code](https://gitlab.psi.ch/H5hut/src/-/commit/c41bd6ec549f1c4e7c90f8d52655054ab56bca57) they select one API or the other depending on HDF5 version. So it should be something like

```suggestion
            and self.spec.satisfies("@:2.0.0rc6")
```

because, if I understand correctly, otherwise `2.0.0rc7` and any other `2.*` version would not build in the future.

Reply:
I think what you say is correct, but although 2.0.0rc7 has the fix in it, the fix isn't present in the autoconf flags in spack. My assumption was that once @2.0.0 is actually released, then the rc1,2,3,4...7 would be removed.
I also assumed that 
```
>>> ver('2.9.9') in ver(':2')
True
```
would return false. I will change the version check to behave correctly

Reply:
The problem is that currently `@2.0.0rc3` is listed, (actually is the only one), and it would stop working since with `@:1` it would build with the new API which does not support. And with `@:2` it would work `@2.0.0rc3`, but nothing newer than `2.0.0rc7`, so also `2.0.0`, `2.0.1`, `2.1`, `2.2`... would not work because spack would configure to use the older API, but the code checks the version of `hdf5` to enable the newly supported API.

Reply:
What about adding the 2.0.0rc7 version as well since the rcs seem to be pretty much regular releases (at least in cadence) anyway? I think then the version constraint by @albestro makes more sense (in practical terms, because there would actually be a version in the package that would not satisfy `@:2.0.0rc6`).

Reply:
Or, if you prefer the `@:1` constraint, which makes sense if/when a non-rc 2.0.0 release exists, then the 2.0.0rc3 version should be removed, or at the very least deprecated?